packages/x-data-grid/src/hooks/utils/useGridVisibleRows.ts

```diff
@@ -36,23 +21,12 @@ export const getVisibleRows = <Api extends GridApiCommon>(
  * - If the row tree has several layers, it contains up to `state.pageSize` top level rows and all their descendants.
  * - If the row tree is flat, it only contains up to `state.pageSize` rows.
  */
+
 export const useGridVisibleRows = <Api extends GridApiCommon>(
   apiRef: React.MutableRefObject<Api>,
-  props: Pick<DataGridProcessedProps, 'pagination' | 'paginationMode'>,
+  // To-do: remove after useGridVisibleRows implementations have been updated
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  props?: Pick<DataGridProcessedProps, 'pagination' | 'paginationMode'>,
 ) => {
-  const response = getVisibleRows(apiRef, props);
-
-  return React.useMemo(
-    () => ({
-      rows: response.rows,
-      range:
-        response.range?.firstRowIndex === undefined || response.range?.lastRowIndex === undefined
-          ? null
-          : {
-              firstRowIndex: response.range?.firstRowIndex ?? 0,
-              lastRowIndex: response.range?.lastRowIndex ?? 0,
-            },
-    }),
-    [response.rows, response.range?.firstRowIndex, response.range?.lastRowIndex],
-  );
+  return useGridSelector(apiRef, gridVisibleRowsSelector, objectShallowCompare);
```

Comment:
Is `objectShallowCompare` really required here? That thing is a hack, in theory it should be possible to avoid it in all cases that use selectors. The only case where that option makes sense is for inline selectors like `useGridSelector(apiRef, () => { ... })` that can't use memoization.

Reply:
You're right, it doesn't, removed.